### ISSUE TITLE
Editorial: Uniform Layout and Naming of Hints, Remarks, Exceptions etc.

### ISSUE THREAD

**Issue by tfrauenstein** [zalando/restful-api-guidelines#660]
We use different layout formats for standard guideline elements like 'Exception:', 'Hint:', 'Note:', 'Remark:', 'Example:', 'Tip', 'Reasoning:', 'Warning:', 'Important:'. Additionally, the elements are partially redundant with unclear 'didactic purpose'. 
Let us reduce the elements to a minimal set with clear purpose and uniform layout. 
It will contribute to readability and efficient consumption by API builders. Examples: 
* Hint: https://opensource.zalando.com/restful-api-guidelines/#101 
* Example: https://opensource.zalando.com/restful-api-guidelines/#116   https://opensource.zalando.com/restful-api-guidelines/#104   https://opensource.zalando.com/restful-api-guidelines/#114
* Note + Tip: https://opensource.zalando.com/restful-api-guidelines/#114
* Reasoning: https://opensource.zalando.com/restful-api-guidelines/#114

**Comment by vadeg** [comment]
I propose to use several features provided by Asciidoc for formatting: 
1. [Admonitions](https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/) for hints, exceptions and etc.
2. [Sidebars](https://docs.asciidoctor.org/asciidoc/latest/blocks/sidebars/) to highlight a block of text.

**Comment by ePaul** [comment]
From the meeting:

* We agree that unifying this is useful
* @vadeg will add a proposal on how to do this (like 3-4 different things)

**Comment by vadeg** [comment]
I have collected a number of occurrences of different elements and mapped them to possible replacements from AsciiDoctor
|Current element | Occurrences|AsciiDoctor Admonition element |
|-|-|-|
|Hint | 32 | TIP or NOTE |
|Example | 10 | Implement new EXAMPLE admonition |
|Note | 47 |  NOTE |
|TIP | 1 | TIP |
|Reasoning | 1 | NOTE | 
|Exception | 7 | IMPORTANT|
|Warning | 1 | WARNING |

**Comment by ePaul** [comment]
@tfrauenstein and @vadeg : please coordinate whether to do this or #674 first, as there will be a lot of conflicts.

**Comment by tkrop** [comment]
It may also make sense to realign on how to emphasize keywords like `MUST`, `SHOULD`, `RECOMMENDED`, etc. In text we currently use **must**, **should**, ... - also often forgetting the bold, while we use MUST, SHOULD, ... in the rule titles. I think we should strive for consistency here too.

**Comment by vadeg** [comment]
@ePaul yes. I am testing the UI for asciidoc. Current schema doesn't support properly some formatting components.

**Comment by vadeg** [comment]
@tfrauenstein 

> a) Let us provide a short definition of when to use TIP vs. NOTE when mapped from Hint

Do you suggest to provide it somewhere in the documen?

**Comment by tfrauenstein** [comment]
@vadeg I do not yet have a specific guidance, and propose that we should proceed only with 'note'. If it turns out that some hints are better handled as 'note', then we can use it and define the criteria. I would not overload the doc with a definition -- it should be clear from context.

**Comment by ePaul** [comment]
We agreed that this is still worth doing, but currently nobody has time.
Approach:
1. consistently use the existing annotations where they fit.
2. use the examples from https://github.com/asciidoctor/asciidoctor-extensions-lab/issues/9 to implement new annotations for exception + example, then use them.

Volunteers are welcome!
